iothub/device/src/Properties.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// A container for properties.
+    /// </summary>
+    public class Properties : IEnumerable<object>
+    {
+        private const string VersionName = "$version";
+        private readonly IDictionary<string, object> _readOnlyProperties = new Dictionary<string, object>();
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="Properties"/>
+        /// </summary>
+        internal Properties()
+        {
+            Writable = new PropertyCollection();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="Properties"/> with the specified collections
+        /// </summary>
+        /// <param name="writablePropertyCollection">A collection of writable properties returned from IoT Hub</param>
+        /// <param name="readOnlyPropertyCollection">A collection of read-only properties returned from IoT Hub</param>
+        internal Properties(PropertyCollection writablePropertyCollection, IDictionary<string, object> readOnlyPropertyCollection)
+        {
+            Writable = writablePropertyCollection;
+            _readOnlyProperties = readOnlyPropertyCollection;
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        public PropertyCollection Writable { get; private set; }
+
+        /// <summary>
+        /// Get the property from the propeties collection
+        /// </summary>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public dynamic this[string propertyName]
```

Comment:
- dynamic vs object vs T
- also make sure that this is in sync with the enumerable return type as well -> however it is implemented

Reply:
Move everything to object to be inline with the enumerator.